src/main/java/com/google/firebase/auth/UserRecord.java

```diff
@@ -204,6 +231,32 @@ private static void checkPassword(String password) {
     checkArgument(password.length() >= 6, "password must be at least 6 characters long");
   }
 
+  private static void checkCustomClaims(Map<String,Object> customClaims) {
+    if (customClaims == null) {
+      return;
+    }
+    for (String key : customClaims.keySet()) {
+      checkArgument(!FirebaseUserManager.RESERVED_CLAIMS.contains(key),
+          "Claim '" + key + "' is reserved and cannot be set");
+    }
+  }
+
+  private static String serializeCustomClaims(Map customClaims, JsonFactory jsonFactory) {
+    checkNotNull(jsonFactory, "JsonFactory must not be null");
+    if (customClaims == null || customClaims.isEmpty()) {
```

Comment:
If you initialize customClaims to an empty map, this would go away as well.

Reply:
This can be a value directly passed in by the user via `UpdateRequest.setCustomClaims()`. Therefore `null` values are allowed here.